### PR TITLE
Fix & update benchmarks and fuzzers

### PR DESCRIPTION
**Pull request by Alan32Liu**
1. Fix and update (to the latest tag/release) the (nontrivial) benchmarks and fuzzers to adapt to `Ubuntu:20.04` and `Python3.10.8`.
2. Delete the trivial/buggy benchmarks and fuzzers.
Base: #1526

# Benchmarks
'[x]' means it can build and run under the new environment.

## Support
* [x] All benchmarks in #1441.
  * [x] [`systemd_fuzz-link-parser`](https://github.com/google/fuzzbench/pull/1541/commits/4ec585808f90d2ce0f6a348815a1af0761ef5c90).
  * [x] [`systemd_fuzz-varlink`](https://github.com/google/fuzzbench/pull/1541/commits/4ec585808f90d2ce0f6a348815a1af0761ef5c90).
  * [x] [`njs_njs_process_script_fuzzer`](https://github.com/google/fuzzbench/pull/1541/commits/5720ea0e26db8459c755b3f9e771b63e2bb090ea).
  * [x] [`libxslt_xpath`](https://github.com/google/fuzzbench/pull/1526/commits/cd60b7efcb2b8281031d5a62ae1471a8b05bd6e5).
  * [x] [`openh264_decoder_fuzzer`](https://github.com/google/fuzzbench/commit/07008f194b7794c5a30b21bc480b3c4c7a79e07a).
  * [x] [`libarchive_libarchive_fuzzer`](https://github.com/google/fuzzbench/commit/da41386fafb75f13c421dbb591e34dee99e2a529).
  * [x] [`ffmpeg_ffmpeg_demuxer_fuzzer`](https://github.com/google/fuzzbench/commit/91d3b2b05694ee492771976b447102c68b9c83de).
    * I `apt install` some packages that were previously built manually, to make its `build.sh` consistent with `OSS-Fuzz`, hope that would be alright.
  * [x] [`mruby-2018-05-23`](https://github.com/google/fuzzbench/commit/53aa92b5f5affaab11ab729f8f0387f8028fe72e).
  * [x] [`openssl_x509`](https://github.com/google/fuzzbench/commit/23e39f1eafa3672a4d822c8c39923e3361baab3f).
  * [x] [`php_php-fuzz-parser-2020-07-25`](https://github.com/google/fuzzbench/commit/52fc22b1e4c8317de5d0bd559141b833c4dd01a6).
  * [x] [`php_php-fuzz-execute`](https://github.com/google/fuzzbench/commit/52fc22b1e4c8317de5d0bd559141b833c4dd01a6).
  * [x] [`zstd_stream_decompress`](https://github.com/google/fuzzbench/commit/57f0d450c4ba5eba7f937987654d3621491ac0d2).
  * [x] [`poppler_pdf_fuzzer`](https://github.com/google/fuzzbench/commit/347523f9d6f1f39f27a7a13875bfa14668513f3d).
  * [x] [`grok_grk_decompress_fuzzer`](https://github.com/google/fuzzbench/commit/fe1d72856fc2cce975c72245d9d1f273f708aef1).

* [x] Bug Benchmarks (temporarily converted to coverage benchmarks):
    * [x] arrow_parquet-arrow-fuzz
    * [x] aspell_aspell_fuzzer
    * [x] ffmpeg_ffmpeg_demuxer_fuzzer
    * [x] file_magic_fuzzer
    * [x] grok_grk_decompress_fuzzer
    * [x] libarchive_libarchive_fuzzer
    * [x] libgit2_objects_fuzzer
    * [x] libhevc_hevc_dec_fuzzer
    * [x] libhtp_fuzz_htp
    * [x] libxml2_libxml2_xml_reader_for_file_fuzzer
    * [x] matio_matio_fuzzer
    * [x] mruby-2018-05-23
    * [x] muparser_set_eval_fuzzer
    * [x] njs_njs_process_script_fuzzer
    * [x] openh264_decoder_fuzzer
    * [x] php_php-fuzz-execute
    * [x] php_php-fuzz-parser-2020-07-25
    * [x] poppler_pdf_fuzzer
    * [x] proj4_standard_fuzzer
    * [x] quickjs_eval-2020-01-05
    * [x] stb_stbi_read_fuzzer
    * [x] systemd_fuzz-varlink
    * [x] usrsctp_fuzzer_connect
    * [x] wireshark_fuzzshark_ip

* [x] Other benchmarks?
  * [x] [`libpng`](https://github.com/google/fuzzbench/commit/66c80b0e6a61254aac732ece72f4f7562660ff39).
  * [x] [`bloaty`](https://github.com/google/fuzzbench/commit/7330b7fc2951a7d1781a13e0e15f90528c028158).

* [ ] Change benchmark dir name to reflect their current version.

## Deprecate
Nothing so far, may add later.

## Undetermined
1. [ ] `poppler`
    * Failed to build in FuzzBench and [OSS-Fuzz](https://github.com/google/oss-fuzz/blob/master/projects/poppler/Dockerfile) (CI cancelled).
    * Can add it back when it can build.
    * Can deprecate if it continues fail to build.

# Fuzzers
Fix and update (to the latest tag/release) the (nontrivial) fuzzers to adapt to `Ubuntu:20.04` and `Python3.10.8`.
Maybe also delete the outdated/trivial/buggy ones?

We want to **support** 
([x] means it is up-to-date and passed the `standard` and `oss-fuzz` categories of CI tests and did not break in the `bug` category before the 5-hour timeout):
* [x] `afl`
* [x] `aflfast`
* [x] `afl++`
* [x] `aflsmart`
* [x] `centipede` (failing due to the weak reference issue, which should be acceptable?)
* [x] `eclipser`
* [x] `fairfuzz`
* [x] `honggfuzz`
* [x] `libafl`
* [x] `libFuzzer`
* [x] `mopt`
* [ ] `klee` (Not sure if we want to remove it. It is actively maintained, but it seems we do not use it in our reports for some reason).
* [ ] `symcc_aflplusplus` (I want to add `SymCC`-related fuzzers, as `SymCC` supports concolic execution and seems impactful. But I failed to fix its errors after trying for two days. `SymCC` still relies on `Clang-10` and `Python2`, yet it also tries to support `LLVM-15` in very recent PRs. Maybe let's wait for a while until it can get rid of the ancient dependencies and becomes stable on `LLVM-15`?)

## Not used as default fuzzers in general comparison evaluations
* [ ] `centipede_function_filter` (Do we use it for daily fuzzing evaluations?)
* [ ] `introspector_driven_focus` (Do we use it for daily fuzzing evaluations?)

## Deprecate
([x] means it is up-to-date and passed the `standard` and `oss-fuzz` categories of CI tests and did not break in the `bug` category before the 5-hour timeout):
* [ ] All variations
* [x] `entropic`
* [ ] `neuzz`
* [ ] `pythia`
* [ ] `fafuzz`
* [ ] `tortoisefuzz`
* [x] `wingfuzz`
* [ ] `weizz`
* [ ] `fuzzolic_aflplusplus_z3`
* [ ] `nautilus`
* [ ] `gramatron`
* [ ] `token_level`
* [ ] `afl_2_52_b`
* [ ] `libfuzzer_dataflow`
* [ ] `lafintel` (Do we want to keep this? It has not been updated in the past 6 years.)

## Undecided
Nothing so far.

### CODE REVIEW

**Comment by Alan32Liu** [comment]
I noticed that we are using [this repo for `SymCC`](https://github.com/AdaLogics/adacc) instead of [the official repo](https://github.com/eurecom-s3/symcc).
Currently, it fails to build and I am planning to fix it.
Shall I continue using that repo or switch to the official one? Thanks! @jonathanmetzman

**Comment by Alan32Liu** [comment]
Which `AFL` shall we keep?
I noticed that we have [`AFL` (which is archived)](https://github.com/google/AFL.git) and [`AFL-2.52b` (which has not been updated in 2 years)](https://github.com/Fuzzers-Archive/afl-2.52b.git).
Shall we keep both of them or stop supporting one of them?

**Comment by Alan32Liu** [comment]
I reckon the `bug` category is still needed by the competition, maybe we exclude them from daily use and CI tests, and only keep them for the competition?

**Comment by jonathanmetzman** [comment]
> I noticed that we are using [this repo for `SymCC`](https://github.com/AdaLogics/adacc) instead of [the official repo](https://github.com/eurecom-s3/symcc). Currently, it fails to build and I am planning to fix it. Shall I continue using that repo or switch to the official one? Thanks! @jonathanmetzman

Just switch it. I think switching make break some variants that use symcc but I think this is fine
CC @DavidKorczynski

**Comment by jonathanmetzman** [comment]
> I reckon the `bug` category is still needed by the competition, maybe we exclude them from daily experiment and CI tests, and only keep them for the competition?
> 
> I've tried to make `bug` benchmarks compatible with `libfuzzer` in `Ubuntu:20.04` and `Python-3.10.8`. CI tests were not able to test them all due to the 5-hour time limit.

Which ones take a long time to finish. Maybe just those should be excluded

**Comment by Alan32Liu** [comment]
> > I noticed that we are using [this repo for `SymCC`](https://github.com/AdaLogics/adacc) instead of [the official repo](https://github.com/eurecom-s3/symcc). Currently, it fails to build and I am planning to fix it. Shall I continue using that repo or switch to the official one? Thanks! @jonathanmetzman
> 
> Just switch it. I think switching make break some variants that use symcc but I think this is fine CC @DavidKorczynski

BTW, what is our plan with the variants?
Would you prefer me to prune some of them (or at least exclude them from CI tests) or also update them?
(e.g., [eclipser_aflplusplus](https://github.com/google/fuzzbench/tree/master/fuzzers/eclipser_aflplusplus), [eclipser_um_prioritize_75](https://github.com/google/fuzzbench/tree/master/fuzzers/eclipser_um_prioritize_75), [libfuzzer_focus_idx0](https://github.com/google/fuzzbench/tree/master/fuzzers/libfuzzer_focus_idx0), and [libfuzzer_dataflow_store](https://github.com/google/fuzzbench/tree/master/fuzzers/libfuzzer_dataflow_store).)

Currently, they do not bother me a lot (except for the noises in CI), but they will affect competition participants.
Personal accounts have a (very low) budget for running CIs, e.g., mine won't last one round of the current CIs.

Maybe we should only run CIs on the fuzzer-benchmark pairs that are involved in each PR?
For example,  a change in benchmark A should not trigger CIs tests on fuzzers that are not supported by A, or any other benchmark programs.

**Comment by Alan32Liu** [comment]
> > I reckon the `bug` category is still needed by the competition, maybe we exclude them from daily experiment and CI tests, and only keep them for the competition?
> > I've tried to make `bug` benchmarks compatible with `libfuzzer` in `Ubuntu:20.04` and `Python-3.10.8`. CI tests were not able to test them all due to the 5-hour time limit.
> 
> Which ones take a long time to finish. Maybe just those should be excluded

I am not sure if this is caused by a specific time-consuming benchmark.
Using `libfuzzer` as an example, it had completed 21 benchmarks in the `bug` category before the 300 min timeout. Each benchmark takes 10-30 minutes, which is not far away from the average value (~14min). The longest is `ffmpeg_ffmpeg_demuxer_fuzzer`, which took roughly 33 minutes, similar to another two benchmarks.
The last one (22nd) `systemd_fuzz-varlink` was only run for 12 minutes before it was killed.

In comparison, `libFuzzer` spent almost 3 hours completing the 11 benchmarks in the `standard` category (which I guess is 'coverage'?), which is close to `bug` on average.

Shall we reduce the number of benchmarks in `bug`, or give it more time?

**Comment by Alan32Liu** [comment]
By the way, shall I clean up the CI workflow, so that we don't have to run so many CI tests and only focus on the important ones?

**Comment by jonathanmetzman** [comment]
> By the way, shall I clean up the CI workflow, so that we don't have to run so many CI tests and only focus on the important ones?

How would you do this? Seems like this would be very involved.

**Comment by jonathanmetzman** [comment]
> > > I noticed that we are using [this repo for `SymCC`](https://github.com/AdaLogics/adacc) instead of [the official repo](https://github.com/eurecom-s3/symcc). Currently, it fails to build and I am planning to fix it. Shall I continue using that repo or switch to the official one? Thanks! @jonathanmetzman
> > 
> > 
> > Just switch it. I think switching make break some variants that use symcc but I think this is fine CC @DavidKorczynski
> 
> BTW, what is our plan with the variants? Would you prefer me to prune some of them (or at least exclude them from CI tests) or also update them? (e.g., [eclipser_aflplusplus](https://github.com/google/fuzzbench/tree/master/fuzzers/eclipser_aflplusplus), [eclipser_um_prioritize_75](https://github.com/google/fuzzbench/tree/master/fuzzers/eclipser_um_prioritize_75), [libfuzzer_focus_idx0](https://github.com/google/fuzzbench/tree/master/fuzzers/libfuzzer_focus_idx0), and [libfuzzer_dataflow_store](https://github.com/google/fuzzbench/tree/master/fuzzers/libfuzzer_dataflow_store).)
> 
> Currently, they do not bother me a lot (except for the noises in CI), but they will affect competition participants. Personal accounts have a (very low) budget for running CIs, e.g., mine won't last one round of the current CIs.

Can you give me a list of ones you want to remove and I'll sign off on them?

> 
> Maybe we should only run CIs on the fuzzer-benchmark pairs that are involved in each PR? For example, a change in benchmark A should not trigger CIs tests on fuzzers that are not supported by A, or any other benchmark programs.

We do this to some extent.

File: benchmarks/arrow_parquet-arrow-fuzz/Dockerfile [hunk 1]
```diff
@@ -28,6 +27,9 @@ RUN apt-get update -y -q && \
         ninja-build \
         python3
 
-RUN git clone --depth=1 https://github.com/apache/arrow.git $SRC/arrow
+RUN git clone \
+        --depth=1 \
+        --branch apache-arrow-10.0.0 \
```
Comment (jonathanmetzman) [inline benchmarks/arrow_parquet-arrow-fuzz/Dockerfile:32]: Sadly I don't think thsi approach youve taken really works with the bug based benchmarks because these benchmarks are only useful when a particular version is checked out. Otherwise they wont have the bugs we expect.

File: benchmarks/bloaty_fuzz_target/Dockerfile [hunk 2]
```diff
@@ -15,7 +15,18 @@
...
-RUN git clone --depth 1 https://github.com/google/bloaty.git bloaty
+
+RUN apt-get update && \
+    apt-get install -y \
+        cmake \
+        ninja-build \
+        g++ \
+        libz-dev
+
+RUN git clone \
```
Comment (jonathanmetzman) [inline benchmarks/bloaty_fuzz_target/Dockerfile:26]: Same issue

File: benchmarks/libpcap_fuzz_both/build.sh [hunk 3]
```diff
@@ -23,12 +23,15 @@ cd build
 cmake ..
 make
 
+# Add required libs for AFL++.
+[[ $FUZZER == aflplusplus ]] && export EXTRA_LIBS='/usr/lib/x86_64-linux-gnu/libdbus-1.a /lib/x86_64-linux-gnu/libsystemd.so.0'
```
Comment (jonathanmetzman) [inline benchmarks/libpcap_fuzz_both/build.sh:27]: This is very bizarre. I'd prefer it if this change could somehow be done in the aflplusplus fuzzer.py instead of here.

**Comment by Alan32Liu** [comment]
> > > > I noticed that we are using [this repo for `SymCC`](https://github.com/AdaLogics/adacc) instead of [the official repo](https://github.com/eurecom-s3/symcc). Currently, it fails to build and I am planning to fix it. Shall I continue using that repo or switch to the official one? Thanks! @jonathanmetzman
> > > 
> > > 
> > > Just switch it. I think switching make break some variants that use symcc but I think this is fine CC @DavidKorczynski
> > 
> > 
> > BTW, what is our plan with the variants? Would you prefer me to prune some of them (or at least exclude them from CI tests) or also update them? (e.g., [eclipser_aflplusplus](https://github.com/google/fuzzbench/tree/master/fuzzers/eclipser_aflplusplus), [eclipser_um_prioritize_75](https://github.com/google/fuzzbench/tree/master/fuzzers/eclipser_um_prioritize_75), [libfuzzer_focus_idx0](https://github.com/google/fuzzbench/tree/master/fuzzers/libfuzzer_focus_idx0), and [libfuzzer_dataflow_store](https://github.com/google/fuzzbench/tree/master/fuzzers/libfuzzer_dataflow_store).)
> > Currently, they do not bother me a lot (except for the noises in CI), but they will affect competition participants. Personal accounts have a (very low) budget for running CIs, e.g., mine won't last one round of the current CIs.
> 
> Can you give me a list of ones you want to remove and I'll sign off on them?

[The top of this PR](https://github.com/google/fuzzbench/pull/1541#issue-1435597783) has three lists: 1. the fuzzers we want to support, the fuzzers we may want to remove, and the fuzzers that I am not sure about (e.g., KLEE).
Would the current categorization look good to you?

File: benchmarks/mruby-2018-05-23/Dockerfile [hunk 4]
```diff
@@ -17,7 +17,10 @@
 FROM gcr.io/oss-fuzz-base/base-builder@sha256:fb1a9a49752c9e504687448d1f1a048ec1e062e2e40f7e8a23e86b63ff3dad7c
 RUN apt-get update && apt-get install -y build-essential ruby bison ninja-build \
     cmake zlib1g-dev libbz2-dev liblzma-dev
-RUN git clone --depth 1 https://github.com/mruby/mruby mruby
+RUN git clone \
```
Comment (jonathanmetzman) [inline benchmarks/mruby-2018-05-23/Dockerfile:20]: CC @andreafioraldi I assume if we upgrade this benchmark we will lose the bugs it was finding?

File: benchmarks/mruby-2018-05-23/Dockerfile [hunk 5]
```diff
@@ -17,7 +17,10 @@
 FROM gcr.io/oss-fuzz-base/base-builder@sha256:fb1a9a49752c9e504687448d1f1a048ec1e062e2e40f7e8a23e86b63ff3dad7c
```
Comment (jonathanmetzman) [inline benchmarks/mruby-2018-05-23/Dockerfile:17]: Realize that if this is pinned to an Ubuntu 16 image then you haven't upgraded the image this is being built in.

File: benchmarks/openh264_decoder_fuzzer/build.sh [hunk 6]
```diff
@@ -20,7 +20,7 @@ svn export https://github.com/mozillasecurity/fuzzdata.git/trunk/samples/h264 co
 mv ./res/*.264 ./corpus/
 zip -j0r ${OUT}/decoder_fuzzer_seed_corpus.zip ./corpus/
 
-# build 
+# build
```
Comment (jonathanmetzman) [inline benchmarks/openh264_decoder_fuzzer/build.sh:23]: Try not to do unrelated changes in a PR like this. It's hard enough to manage as is.

File: benchmarks/php_php-fuzz-execute/Dockerfile [hunk 7]
```diff
@@ -17,7 +17,14 @@
 FROM gcr.io/oss-fuzz-base/base-builder@sha256:fb1a9a49752c9e504687448d1f1a048ec1e062e2e40f7e8a23e86b63ff3dad7c
 RUN apt-get update && \
     apt-get install -y autoconf automake libtool bison re2c pkg-config
-RUN git clone --depth 1 --branch master https://github.com/php/php-src.git php-src
-RUN git clone https://github.com/kkos/oniguruma.git php-src/oniguruma
+RUN git clone \
```
Comment (jonathanmetzman) [inline benchmarks/php_php-fuzz-execute/Dockerfile:20]: The correct way to pin these benchmarks is here: https://github.com/google/fuzzbench/blob/master/benchmarks/php_php-fuzz-execute/benchmark.yaml#L1

If none of the bug based benchmarks upgrade (which is possible) I'm happy to junk them.

**Review by jonathanmetzman — COMMENTED** [review]
I think these upgrades weren't done in an ideal way due to some of the complexity of fuzzbench. Let's discuss this.

File: benchmarks/arrow_parquet-arrow-fuzz/Dockerfile (hunk 1 above)
Comment (Alan32Liu) [reply benchmarks/arrow_parquet-arrow-fuzz/Dockerfile:32]: After our discussion today, I was thinking about the following, but not sure if it would be a good idea:
1. We build **all** benchmarks **only** with the `Dockerfile` for coverage experiments.
2. If the benchmark specifies a commit in its config file, we first build it with the `Dockerfile` and then checkout to the commit, as use it for bug experiments.

In that way, we can use all bug benchmarks for coverage (we have already supported them anyway), and the commit in the config file is no longer a redundant design.

File: benchmarks/libpcap_fuzz_both/build.sh (hunk 3 above)
Comment (Alan32Liu) [reply benchmarks/libpcap_fuzz_both/build.sh:27]: Yeah, I was thinking about the same trade-off.
I put it here because I assumed that it is more likely that 1) other fuzzers will need this on this benchmark, and less likely that 2) `aflp++` will need this on a different benchmark.

For 1), we can simply add another fuzzers in the first operand of `&&`.
For 2), yes, it is better to put the line in `afl++`.

File: benchmarks/mruby-2018-05-23/Dockerfile (hunk 5 above)
Comment (Alan32Liu) [reply benchmarks/mruby-2018-05-23/Dockerfile:17]: Sorry, I am not sure if I understood: Why is it pinned to 16?
The hash is `Ubuntu 20.04.5 LTS`.

File: benchmarks/mruby-2018-05-23/Dockerfile (hunk 4 above)
Comment (andreafioraldi) [reply benchmarks/mruby-2018-05-23/Dockerfile:20]: yes ofc this is a vulnerable version

File: benchmarks/mruby-2018-05-23/Dockerfile (hunk 5 above)
Comment (jonathanmetzman) [reply benchmarks/mruby-2018-05-23/Dockerfile:17]: Nevermind youre right.

**Comment by jonathanmetzman** [comment]
I guess I'm OK with not fixing anything but the core fuzzers.

File: benchmarks/arrow_parquet-arrow-fuzz/Dockerfile (hunk 1 above)
Comment (Alan32Liu) [reply benchmarks/arrow_parquet-arrow-fuzz/Dockerfile:32]: Ah sorry, 1 means not using the commit hash in `project.yaml`. It will still need `build.sh`.

The intention is to make use of current bug benchmark programs (especially the ones that are upgradable) for code-coverage evaluation, and the commit in `project.yaml` for bug-finding evaluation.

For example, say we have a bug benchmark `B` that defines a commit in its `project.yaml`:
1. We first use it as a coverage benchmark:
    * We build it with its `Dockerfile` and `build.sh` as usual but do not checkout to the commit in `project.yaml`.
3. We also use it as a bug benchmark:
    * We build it with its `Dockerfile` and `build.sh` as usual and then checkout to the commit in `project.yaml`.

This will not work for some bug benchmarks and that is fine: We only remove those and keep the ones that work.

Comment (Alan32Liu) [reply benchmarks/arrow_parquet-arrow-fuzz/Dockerfile:32]: OK, would it be a good idea to have a coverage benchmark directory for `B` (e.g. `benchmarks/B/`) and another separate bug benchmark directory for the buggy commit of `B`(e.g. `benchmark/B_<commit/date/version>`)?

The downside is having to tolerate code duplication in their `Dockerfile` and `build.sh`.

File: benchmarks/libpcap_fuzz_both/build.sh (hunk 3 above)
Comment (Alan32Liu) [reply benchmarks/libpcap_fuzz_both/build.sh]: SG! Moved to `afl++`'s `fuzzer.py`.

File: benchmarks/arrow_parquet-arrow-fuzz/Dockerfile (hunk 1 above)
Comment (Alan32Liu) [reply benchmarks/arrow_parquet-arrow-fuzz/Dockerfile:32]: Based on our discussion today, I plan to proceed with the following steps:
1. Step 1: Treat all benchmarks are coverage benchmarks so that participants can test building their fuzzers with them.
    * Also upgrade bug benchmarks to the latest version (which is bug-free) so that participants can test them under the same environment.
2. Step 2: Temporarily remove the 1) commit hash and 2) `type: bug` label from their `benchmark.yaml` file.
3. Step 3: Find 5 benchmarks whose old buggy hash commit can work under the new environment.
4. Step 4: Add their corresponding 1) commit hash and 2) `type: bug` label back to their `benchmark.yaml` file.
5. Step 5: Find another 20 bug benchmarks.

Now:
1. [x] Step 1: We already did it in this PR.
2. [ ] Step 2: I will proceed if it looks good to you.
3. [ ] Step 3-5: These can be done later after we merge #1526 to the master branch.

Would this be a good plan to proceed? @jonathanmetzman 
Thanks!

Comment (jonathanmetzman) [reply benchmarks/arrow_parquet-arrow-fuzz/Dockerfile:32]: Sure!

**Comment by Alan32Liu** [comment]
Interesting. [`libxml2`](https://gitlab.gnome.org/GNOME/libxml2) is not available.
Some benchmarks are broken because of it.

**Comment by Alan32Liu** [comment]
The general rule I followed is to consider `libFuzzer` as the default fuzzer and `libpng-1.6.38` as the default benchmark.
If a benchmark `B` can build with `libFuzzer` then we can keep `B`, and if a fuzzer `A` failed to build with `B`, then we add `A` to the list of `unsupported_fuzzers` in `B`'s `benchmark.yaml`.
Similarly, if a fuzzer `A` can build with `libpng-1.6.38`, then we keep `A`, and if `A` cannot build with another benchmark `B`, then we add `A` to the list of `unsupported_fuzzers` in `B`'s `benchmark.yaml`.

In addition, we deprecate `A` is `A` is no longer maintained or relies on very old dependencies.

**Comment by vanhauser-thc** [comment]
> Which `AFL` shall we keep? I noticed that we have [`AFL` (which is archived)](https://github.com/google/AFL.git) and [`AFL-2.52b` (which has not been updated in 2 years)](https://github.com/Fuzzers-Archive/afl-2.52b.git). Shall we keep both of them or stop supporting one of them?

IMHO the archive one should be kept because still people like to reference against it. 2.52b should be dropped though.

**Comment by Alan32Liu** [comment]
> > Which `AFL` shall we keep? I noticed that we have [`AFL` (which is archived)](https://github.com/google/AFL.git) and [`AFL-2.52b` (which has not been updated in 2 years)](https://github.com/Fuzzers-Archive/afl-2.52b.git). Shall we keep both of them or stop supporting one of them?
> 
> IMHO the archive one should be kept because still people like to reference against it. 2.52b should be dropped though.

Yep, thanks!
We think the same: This PR keeps `AFL` and drops `AFL-2.52b` : )

Are there any other fuzzers that you would suggest keeping/dropping?

**Comment by vanhauser-thc** [comment]
I think lafintel, mopt, aflfast and fairfuzz can be dropped. aflsmart is also pretty outdated, but its one of the few structured fuzzing implementations so that is why I would keep it.

**Comment by jonathanmetzman** [comment]
> Interesting. GitLab [`GNOME`](https://gitlab.gnome.org/GNOME/libxslt/-/wikis/home) packages (e.g. [`libxml2`](https://gitlab.gnome.org/GNOME/libxml2), [`libxslt`](https://gitlab.gnome.org/GNOME/libxslt/-/wikis/home)) are not available. Our benchmarks are broken because of it.

Let me close and reopen and see if that fixes things

**Comment by Alan32Liu** [comment]
> > Interesting. GitLab [`GNOME`](https://gitlab.gnome.org/GNOME/libxslt/-/wikis/home) packages (e.g. [`libxml2`](https://gitlab.gnome.org/GNOME/libxml2), [`libxslt`](https://gitlab.gnome.org/GNOME/libxslt/-/wikis/home)) are not available. Our benchmarks are broken because of it.
> 
> Let me close and reopen and see if that fixes things

Ah sorry, that was fixed automatically on the following day.
They were only down for one night.

**Comment by Alan32Liu** [comment]
> I think lafintel, mopt, aflfast and fairfuzz can be dropped. aflsmart is also pretty outdated, but its one of the few structured fuzzing implementations so that is why I would keep it.

Thanks!
Yeah, `lafintel` has not been maintained for years, I excluded it in this PR.
I will have another PR dedicated to excluding more of those fuzzers (and benchmarks) later.

**Comment by Alan32Liu** [comment]
OK, I will merge this back to its base branch as discussed.
Will have another PR to remove redundant fuzzers/benchmarks.
